contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -183,32 +214,168 @@ contract MixinSignatureValidator is
             signerAddress,
             signature
         );
-        if (signatureType == SignatureType.OrderValidator) {
-            // The entire order is verified by validator contract.
-            isValid = _validateOrderWithValidator(
-                order,
+        if (signatureType == SignatureType.Validator) {
+            // The entire order is verified by a validator contract.
+            isValid = _validateBytesWithValidator(
+                abi.encode(order, orderHash),
```

Comment:
I'm wondering if it still makes sense to pass the `orderHash` in here. Would you blindly trust that the hash corresponds to the data when verifying the signature? I feel like if you _only_ want the hash, you'd probably just use the regular `Wallet` type. If you want both the hash and the data, you probably won't trust the hash without verifying.

Reply:
I also feel like the `Validator` type isn't very useful with just a hash, generally speaking.

Reply:
My arguments for including the hash would be that the hash is a very useful, canonical ID, which is not cheap to compute. For example, a validator could use it to immediately look up the `filled` state of the order.

Also, wouldn't just gating your callback such that `msg.sender == exchange` be sufficient to trust that the hash provided is authentic?

Reply:
I suppose it can't hurt. Curious to see how much it gets used.